homeassistant/components/onkyo/media_player.py

```diff
@@ -349,242 +411,174 @@ def unique_id(self):
-            if source in self._source_mapping:
-                self._current_source = self._source_mapping[source]
-                break
-            self._current_source = "_".join(current_source_tuples[1])
-        self._muted = bool(mute_raw[1] == "on")
-        if preset_raw and self._current_source.lower() == "radio":
-            self._attributes[ATTR_PRESET] = preset_raw[1]
-        elif ATTR_PRESET in self._attributes:
-            del self._attributes[ATTR_PRESET]
-        if self._supports_volume:
-            # AMP_VOL/MAX_RECEIVER_VOL*(MAX_VOL/100)
-            self._volume = (
-                volume_raw[1] / self._receiver_max_volume * (self._max_volume / 100)
+    def _query_delayed_av_info(self):
+        if self._zone == "main" and not self._query_timer:
+            self._query_timer = self.hass.loop.call_later(
+                AUDIO_VIDEO_INFORMATION_UPDATE_INTERVAL, self._query_av_info
             )
 
-    @property
-    def supported_features(self):
-        """Return media player features that are supported."""
-        if self._supports_volume:
-            return SUPPORT_ONKYO
-        return SUPPORT_ONKYO_WO_VOLUME
-
-    def turn_off(self):
-        """Turn the media player off."""
-        self.command(f"zone{self._zone}.power=standby")
-
-    def set_volume_level(self, volume):
-        """
-        Set volume level, input is range 0..1.
-
-        However full volume on the amp is usually far too loud so allow the user to specify the upper range
-        with CONF_MAX_VOLUME.  we change as per max_volume set by user. This means that if max volume is 80 then full
-        volume in HA will give 80% volume on the receiver. Then we convert
-        that to the correct scale for the receiver.
-        """
-        # HA_VOL * (MAX VOL / 100) * MAX_RECEIVER_VOL
-        self.command(
-            f"zone{self._zone}.volume={int(volume * (self._max_volume / 100) * self._receiver_max_volume)}"
-        )
-
-    def volume_up(self):
-        """Increase volume by 1 step."""
-        self.command(f"zone{self._zone}.volume=level-up")
-
-    def volume_down(self):
-        """Decrease volume by 1 step."""
-        self.command(f"zone{self._zone}.volume=level-down")
-
-    def mute_volume(self, mute):
-        """Mute (true) or unmute (false) media player."""
-        if mute:
-            self.command(f"zone{self._zone}.muting=on")
-        else:
-            self.command(f"zone{self._zone}.muting=off")
-
-    def turn_on(self):
-        """Turn the media player on."""
-        self.command(f"zone{self._zone}.power=on")
-
-    def select_source(self, source):
-        """Set the input source."""
-        if source in self._source_list:
-            source = self._reverse_mapping[source]
-        self.command(f"zone{self._zone}.selector={source}")
+    @callback
+    def _query_av_info(self):
+        self._query_avr("audio-information")
+        self._query_avr("video-information")
```

Comment:
We'll need to ensure those queries (here and in `backfill_state`) don't run they fail once. See https://github.com/home-assistant/core/issues/46130

Reply:
On lines 540 and 554 we already check if the received value equals "N/A", is this not enough?

Reply:
Not really. The idea is that if it fails once, we shouldn't query again.

Reply:
But A/V info isn't always pushed, so we also query it manually in certain cases.

Reply:
You're right, it's been a while 😋
I can make something for it. Is one check for both audio and video info enough, or should we separate them?

Reply:
I don't know - my receiver doesn't have his problem..